experimental/CliffordOrders/src/CliffordOrders.jl

```diff
@@ -0,0 +1,1056 @@
+    coeff_res[i] #Throw a BoundsError for consistency
+  end
+  coeff_res[2^(i - 1) + 1] = QQ(1)
+  return C(coeff_res)
+end
+
+@doc raw"""
+    basis(C::ZZCliffordOrder) -> Vector{ZZCliffordOrderElem}
+
+Return the canonical basis of the Clifford order $C$.
+"""
+basis(C::ZZCliffordOrder) = map(i -> basis(C, i), 1:rank(C))
+
+@doc raw"""
+    gens(C::ZZCliffordOrder) -> Vector{ZZCliffordOrderElem}
+
+Return the canonical multiplicative generating set of the Clifford order $C$.
+"""
+gens(C::ZZCliffordOrder) = map(i -> gen(C, i), 1:rank(lattice(C)))
+
+@doc raw"""
+    is_commutative(C::ZZCliffordOrder) -> Bool
+
+Return `true` if $C$ is commutative and `false` otherwise.
+"""
+is_commutative(C::ZZCliffordOrder) = rank(C) == 1 || rank(C) == 2
+
+################################################################################
+#
+#  Element access
+#
+################################################################################
+
+@doc raw"""
+    coeff(x::CliffordOrderElem, i::Int) -> RingElem
+
+Return the `i`-th coefficient of the element `x`.
+"""
+coeff(x::CliffordOrderElem, i::Int) = coefficients(x)[i]
+
+getindex(x::CliffordOrderElem, i::Int) = coefficients(x)[i]
+
+@doc raw"""
+    coeff(x::ZZCliffordOrderElem, i::Int) -> ZZRingElem
+
+Return the `i`-th coefficient of the element `x`.
+"""
+coeff(x::ZZCliffordOrderElem, i::Int64) = coefficients(x)[i]
+
+getindex(x::ZZCliffordOrderElem, i::Int64) = coefficients(x)[i]
+
+################################################################################
+#
+#  Other functionality
+#
+################################################################################
+
+@doc raw"""
+    centroid(C::CliffordOrder) -> PMat{NumFieldElem, NumFieldOrderFractionalIdeal}
+
+Return the centroid of $C$. Unless `rank(lattice(C)) = 0`, it is a lattice of rank two,
+so it is returned as a pseudo-matrix with two rows. The first row is the coefficient
+vector of the multiplicative identity of $C$ with coefficients in `base_ring(C)`.
+"""
+function centroid(C::CliffordOrder)
+  if isdefined(C, :centroid)
+    return C.centroid
+  end
+  _set_centroid_and_disq!(C)
+  return C.centroid
+end
+
+@doc raw"""
+    quadratic_discriminant(C::CliffordOrder) -> Tuple{NumFieldOrderFractionalIdeal, NumFieldElem}
+
+Return the quadratic discriminant of $C$.
+"""
+function quadratic_discriminant(C::CliffordOrder) 
+  if isdefined(C, :disq)
+    return C.disq
+  end
+  _set_centroid_and_disq!(C)
+  return C.disq
+end
+
+@doc raw"""
+    disq(C::CliffordOrder) -> Tuple{NumFieldOrderFractionalIdeal, NumFieldElem}
+
+Alias for `quadratic_discriminant`.
+"""
+disq(C::CliffordOrder) = quadratic_discriminant(C)
+
+@doc raw"""
+    center(C::CliffordOrder) -> PMat{NumFieldElem, NumFieldOrderFractionalIdeal}
+
+Return the center of $C$. It equals `centroid(C)`, if and only if `rank(lattice(C))`
+is odd. Otherwise it free of rank one, generated by the multiplicative identity of $C$.
+"""
+function center(C::CliffordOrder)
```

Comment:
As above for the algebra.

Reply:
Renamed to pseudo_basis_of_center in this [commit](https://github.com/oscar-system/Oscar.jl/pull/4370/commits/d4b78c05016fbec7d6ab36f6b18b4b8bec2d084d)